src/allmydata/storage/server.py

```diff
@@ -699,6 +690,24 @@ def _allocate_slot_share(self, bucketdir, secrets, sharenum,
                                          self)
         return share
 
+    def list_mutable_shares(self, storage_index) -> set[int]:
+        """List all share numbers for the given mutable.
+
+        Raises ``KeyError`` if the storage index is not known.
+        """
+        si_dir = storage_index_to_dir(storage_index)
+        # shares exist if there is a file for them
+        bucketdir = os.path.join(self.sharedir, si_dir)
+        if not os.path.isdir(bucketdir):
+            raise KeyError("Not found")
```

Comment:
I don't know how strong the guarantee is right now that if there are no shares in a bucket then the bucket directory will not even exist.

So I think it might be possible that sometimes code hits this path and gets a KeyError and other times it proceeds but ultimately returns an empty set.

Also, even if the bucket exists, I suppose it could contain only non-share files - also leading to an empty set result.

Maybe all of the empty set cases could be represented the same way - either w/ KeyError or as an empty set.

Reply:
I'll go with empty set.